rfc1035label/label_test.go

```diff
@@ -52,15 +42,85 @@ func TestLabelToBytes(t *testing.T) {
 		0x2, 'i', 't',
 		0x0,
 	}
-	if !bytes.Equal(encodedLabel, expected) {
-		t.Fatalf("Invalid label. Expected: %v, got: %v", expected, encodedLabel)
+	require.Equal(t, expected, encodedLabel)
+}
+
+func TestLabelsToBytes(t *testing.T) {
+	expected := []byte{
+		9, 's', 'l', 'a', 'c', 'k', 'w', 'a', 'r', 'e',
+		2, 'i', 't',
+		0,
+		9, 'i', 'n', 's', 'o', 'm', 'n', 'i', 'a', 'c',
+		9, 's', 'l', 'a', 'c', 'k', 'w', 'a', 'r', 'e',
+		2, 'i', 't',
+		0,
 	}
+	encodedLabels := LabelsToBytes([]string{"slackware.it", "insomniac.slackware.it"})
+	require.Equal(t, expected, encodedLabels)
 }
 
 func TestLabelToBytesZeroLength(t *testing.T) {
 	encodedLabel := LabelToBytes("")
 	expected := []byte{0}
-	if !bytes.Equal(encodedLabel, expected) {
-		t.Fatalf("Invalid label. Expected: %v, got: %v", expected, encodedLabel)
+	require.Equal(t, expected, encodedLabel)
+}
+
+func TestCompressedLabel(t *testing.T) {
+	data := []byte{
+		// slackware.it
+		9, 's', 'l', 'a', 'c', 'k', 'w', 'a', 'r', 'e',
+		2, 'i', 't',
+		0,
+		// insomniac.slackware.it
+		9, 'i', 'n', 's', 'o', 'm', 'n', 'i', 'a', 'c',
+		192, 0,
+		// mail.systemboot.org
+		4, 'm', 'a', 'i', 'l',
+		10, 's', 'y', 's', 't', 'e', 'm', 'b', 'o', 'o', 't',
+		3, 'o', 'r', 'g',
+		0,
+		// systemboot.org
+		192, 31,
+	}
+	expected := []string{
+		"slackware.it",
+		"insomniac.slackware.it",
+		"mail.systemboot.org",
+		"systemboot.org",
+	}
+
+	labels, err := LabelsFromBytes(data)
+	require.NoError(t, err)
+	require.Equal(t, expected, labels)
+}
+
+func TestShortCompressedLabel(t *testing.T) {
+	data := []byte{
+		// slackware.it
+		9, 's', 'l', 'a', 'c', 'k', 'w', 'a', 'r', 'e',
+		2, 'i', 't',
+		0,
+		// insomniac.slackware.it
+		9, 'i', 'n', 's', 'o', 'm', 'n', 'i', 'a', 'c',
+		192,
+	}
+
+	_, err := LabelsFromBytes(data)
```

Comment:
Should the return value also be tested?

Reply:
not in this case - the function returns an error from the invalid data, no valid data is expected to be returned

Reply:
Oh right